crates/steel-parser/src/ast.rs

```diff
@@ -977,6 +977,8 @@ pub struct List {
 impl PartialEq for List {
     fn eq(&self, other: &Self) -> bool {
         self.args == other.args
+            && self.improper == other.improper
+            && self.location == other.location
```

Comment:
I'm making list less special, and instead rely on an `Eraser` visitor to help in tests.